pkg/apis/core/types_shoot.go

```diff
@@ -83,7 +83,7 @@ type ShootSpec struct {
 	// SeedName is the name of the seed cluster that runs the control plane of the Shoot.
 	SeedName *string
 	// SeedSelector is an optional selector which must match a seed's labels for the shoot to be scheduled on that seed.
-	SeedSelector *metav1.LabelSelector
+	SeedSelector *SeedSelector
```

Comment:
As discussed, I would not change this field for now and only limit the adaptation of the seed selector to the `CloudProfile` (unless there is a good reason to do it also in the `Shoot`).

Reply:
Why do you want a different behaviour for the seed selector for shoots and profiles? I makes sense to allow shoots to additionally restrict seeds according to their provider type.

Reply:
Hm, okay, agreed, I don't have a good argument against it. Let's keep it then.